extensions/amp-youtube/0.1/amp-youtube.js

```diff
@@ -71,6 +91,57 @@ class AmpYoutube extends AMP.BaseElement {
     // the playback.
     return false;
   }
+
+  /** @private */
+  buildImagePlaceholder_() {
+    const imgPlaceholder = new Image();
+    const videoid = this.videoid_;
+
+    // Because sddefault.jpg isn't available for all videos, we try to load
+    // it and fallback to hqdefault.jpg.
+    loadPromise(imgPlaceholder).then(function waitTillReady() {
+      // Wait until the image has completed.
+      if (!imgPlaceholder.complete) {
+        return timer.promise(10).then(waitTillReady);
```

Comment:
Let's try one more thing. Let's wait for the DOMReady as we just tried and then add 10ms on top of it. We may need to cycle through the whole event loop before complete is set. This would still allow us to avoid waiting for undefined time/poll.